### PR TITLE
fix: unmapped `DBT_PROJECT_DIR` env vars for dbt setting

### PR DESCRIPTION
**Pull request by aaronsteers**
This is probably an issue for other settings and other dbt-ext-based utilities.

Each setting that `dbt-ext` should explicitly use an `env: ` that is mapped to what `dbt-ext` expects.

I added this for `DBT_PROJECT`, and confirmed locally in my own project that it was not receiving the updated var prior to this change. There are other configs that likely are also not being mapped correctly. For instance, `path` doesn't look like it is getting mapped anywhere.

Here's the code where `DBT_PROJECT_DIR` and `DBT_PROFILES_DIR` are expected in `dbt-ext`:

https://github.com/meltano/dbt-ext/blob/a53bf00b766190d092f07840c047b51db08f6d1f/dbt_ext/extension.py#L43-L46

### CODE REVIEW

File: _data/meltano/utilities/dbt-duckdb/jwills.yml [hunk 1]
```diff
@@ -86,6 +86,7 @@ settings:
   value: duckdb
 - label: Projects Directory
   name: project_dir
+  env: DBT_PROJECT_DIR
```
Comment (aaronsteers) [inline _data/meltano/utilities/dbt-duckdb/jwills.yml:89]: Reorder to make the linter happy.

```suggestion
- env: DBT_PROJECT_DIR
  label: Projects Directory
  name: project_dir  
```

**Comment by aaronsteers** [comment]
pre-commit.ci run

File: _data/meltano/utilities/dbt-duckdb/jwills.yml [hunk 2]
```diff
@@ -84,8 +84,9 @@ settings:
   label: dbt Profile type
   name: type
   value: duckdb
-- label: Projects Directory
-  name: project_dir
+- env: DBT_PROJECT_DIR
+  label: Projects Directory
+  name: project_dir
```
Comment (aaronsteers) [inline _data/meltano/utilities/dbt-duckdb/jwills.yml:89]:
```suggestion
  name: project_dir
```

**Review by tayloramurphy — APPROVED** [review]

**Comment by pnadolny13** [comment]
@tayloramurphy @aaronsteers is this good to merge?

**Comment by tayloramurphy** [comment]
@pnadolny13 this seems ok to my eyes. Not sure what else AJ needs on this PR though.

**Comment by aaronsteers** [comment]
Sorry - Working on other priorities, I lost track of this PR.
 
Merging now.

**Pull request merged**